solr/core/src/java/org/apache/solr/cloud/OverseerTaskProcessor.java

```diff
@@ -466,6 +466,8 @@ private void markTaskComplete(String id, String asyncId)
           log.warn("Could not find and remove async call [" + asyncId + "] from the running map.");
         }
       }
+
+      workQueue.remove(head);
```

Comment:
@markrmiller can you think of any reason not to do this?  I don't understand why currently getting things out of the queue takes an extra iteration.  I think my fix unmasked a latent problem exposed by DeleteStatusTest; to get that test to pass I have to eagerly remove completed items from the work queue, which seems correct to me.  Not sure why we'd want to wait for a loop-around to `cleanUpWorkQueue()`